core/rawdb/celo_accessors.go

```diff
@@ -0,0 +1,37 @@
+package rawdb
+
+import (
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/ethdb"
+)
+
+var (
+	CeloPreGingerbreadBaseFeePrefix = []byte("celoPgbBlockBaseFee-") // CeloPreGingerbreadBlockBaseFeePrefix + block hash -> BaseFee
+)
+
+type PreGingerbreadFields struct {
+	BaseFee  *big.Int
+	GasLimit *big.Int
+}
+
+// preGingerbreadBlockBaseFeeKey calculates a database key of pre-Gingerbread block BaseFee for the given block hash
+func preGingerbreadBlockBaseFeeKey(hash common.Hash) []byte {
+	return append(CeloPreGingerbreadBaseFeePrefix, hash[:]...)
+}
+
+// ReadPreGingerbreadBlockBaseFee reads BaseFee of pre-Gingerbread block from the given database for the given block hash
+func ReadPreGingerbreadBlockBaseFee(db ethdb.KeyValueReader, blockHash common.Hash) (*big.Int, error) {
+	data, _ := db.Get(preGingerbreadBlockBaseFeeKey(blockHash))
+	if len(data) == 0 {
+		return nil, nil
+	}
```

Comment:
I think it would be better to not ignore the error here, I know that in general the go-ethereum codebase ingores errors from `db.Get` but I think that is probably not a good approach. Given that this function returns an error and you handle it in other parts of the code, I would also return the error here.
```suggestion
	data, _ := db.Get(preGingerbreadBlockBaseFeeKey(blockHash))
	if err != nil {
			return nil, fmt.Errorf("error retrieving pre gingerbread base fee for block: %s, error: %w", blockHash, err)
	if len(data) == 0 {
		return nil, nil
	}
```

Reply:
Yes, agree with that. As you mentioned, I initially followed how geth handles errors in database. However, it's generally better for errors to be handled by the caller for greater flexibility.

Fixed at dac08dbd886c9674fc6fc55499a45986ff007274